aten/src/ATen/native/cpu/radix_sort.h

```diff
@@ -0,0 +1,194 @@
+#pragma once
+#include <ATen/Config.h>
+
+#if !AT_PARALLEL_OPENMP
+
+namespace at { namespace native {
+
+bool inline is_radix_sort_available() { return false; }
+
+template <typename K, typename V>
+std::pair<K*, V*> radix_sort_parallel(
+    K* inp_key_buf,
+    V* inp_value_buf,
+    K* tmp_key_buf,
+    V* tmp_value_buf,
+    int64_t elements_count,
+    int64_t max_value) {
+  TORCH_CHECK(false, "radix_sort_parallel: ATen is not compiled with OpenMP support");
+}
+
+}} // at::native
+
+#else
+
+#include <omp.h>
+#include <c10/util/llvmMathExtras.h>
+
+namespace at { namespace native {
+
+namespace {
+
+// `radix_sort_parallel` is primarily used for converting COO to CSR when sorting
+// the indices, which is used in scatter_reduce optimization on CPU.
+//
+// Copied from fbgemm implementation here:
+// https://github.com/pytorch/FBGEMM/blob/main/fbgemm_gpu/src/cpu_utils.cpp
```

Comment:
We depend on FBGEMM, why do we need to copy paste it here?

Reply:
the problem is that this parallel radix sort is under `fbgemm_gpu` which will be disabled for CPU only build.
if we can move `radix sort` out of `fbgemm_gpu`, say put it under a public common util, i am happy to replace this header.
is it OK to put this on TODO ?

Reply:
ok, this is fine for now, but can we put in a request in fbgemm to move this from gpu to regular fbgemm so that it is available in cpu only too

Reply:
put this as TODO.

Reply:
Can you please reference an issue or something in the comment?